app/src/main/java/cz/martykan/forecastie/activities/MainActivity.java

```diff
@@ -354,8 +364,10 @@ private ParseResult parseTodayJson(String result) {
 
             JSONObject coordinates = reader.getJSONObject("coord");
             if (coordinates != null) {
+                todayWeather.setLat(coordinates.getDouble("lat"));
+                todayWeather.setLon(coordinates.getDouble("lon"));
                 SharedPreferences sp = PreferenceManager.getDefaultSharedPreferences(this);
-                sp.edit().putFloat("latitude", (float) coordinates.getDouble("lon")).putFloat("longitude", (float) coordinates.getDouble("lat")).commit();
+                sp.edit().putFloat("latitude", (float) todayWeather.getLat()).putFloat("longitude", (float) todayWeather.getLon()).commit();
```

Comment:
I think that the latitude and longitude no longer have to be stored in the shared preferences. They are now part of `todayWeather` so they should be available through that wherever they are needed. But you may want to get some feedback from someone more familiar with the codebase for this, or test it out yourself.